### ISSUE TITLE
handleResize should not call setState when unmounted

### ISSUE THREAD

**Issue by schmitzc** [xiaolin/react-image-gallery#498]
I'm getting this console warning: 

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in %s.%sthe componentWillUnmount method
    in ImageGallery (created by Context.Consumer)
```

Tracked it down to the `handleResize` method calling `setState`.

**Comment by xiaolin** [comment]
What are the steps to reproduce this? Can you reproduce with a clean install of the lib?

**Comment by schmitzc** [comment]
> What are the steps to reproduce this? Can you reproduce with a clean install of the lib?

We are encountering this issue in our end-to-end tests. The test navigates to a page with an image gallery, and the issue occurs when it navigates to a new page and the image gallery is unmounted. We had to add a short sleep to wait for the `handleResize` function to finish executing before navigating to a new page.

**Comment by pa-ulander** [comment]
Hi, just to confirm, I have encountered the same issue. Cheers!

**Comment by xiaolin** [comment]
Fixed and released in v1.0.10 https://github.com/xiaolin/react-image-gallery/releases/tag/v1.0.10
